test/storage/varlen_pool_test.cpp

```diff
@@ -0,0 +1,45 @@
+//===----------------------------------------------------------------------===//
```

Comment:
Maybe you want to write more tests. Even though the implementation may change soon, these tests are not implementation specific and can be reused.

At least test for concurrency.